olaf/__main__.py

```diff
@@ -0,0 +1,114 @@
+#:/usr/bin/python
+import os
+import re
+import sys
+import argparse
+import importlib.util
+
+
+def list_pipeline_names(module_name):
+    # Import the module
+    spec = importlib.util.find_spec(module_name)
+    if spec is None:
+        print("Module not found")
+        return
+
+    module = importlib.util.module_from_spec(spec)
+    spec.loader.exec_module(module)
+
+    # Get the directory containing the module
+    module_dir = os.path.dirname(module.__file__)
+
+    # List files in the module directory
+    pipelines = []
+    for filename in os.listdir(module_dir):
+        # Check if the file is a Python module and doesn't match the pattern
+        if (
+            filename.endswith(".py")
+            and not re.match(r"(__\w+__|_\w+)\.py", filename)
+            and filename[:-3] != "runner"
+        ):
+            module_name = filename[:-3]  # Remove the ".py" extension
+        pipelines.append(module_name)
```

Comment:
what is the point of lines 24 to 32 if line 34 overwrite it ?

Reply:
I retrieve automatically all files in the `scripts` folder excepted the `runner` file each file is a demo pipeline.
So to create a new pipeline , we have to create a file , in this file create a new subclass of `Runner` class and then implement the `add_pipeline_components` and `run` methods.